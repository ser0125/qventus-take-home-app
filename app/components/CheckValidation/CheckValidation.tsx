import checkmark from "../../assets/checkmark.svg";
import xmark from "../../assets/x-white.svg";
import Image from "next/image";

const CheckValidation = ({
  label,
  isValid,
}: {
  label: string;
  isValid: boolean;
}) => {
  return (
    <div className="flex items-center">
      <div
        className={`w-10 h-10 rounded-full mx-2.5 my-1 flex justify-center ${
          isValid ? "bg-lime-700" : "bg-red-700"
        }`}
      >
        {isValid ? (
          <Image className="w-5" src={checkmark} alt="check_mark" />
        ) : (
          <Image className="w-5" src={xmark} alt="x_mark" />
        )}
      </div>
      <span>{label}</span>
    </div>
  );
};

export default CheckValidation;
