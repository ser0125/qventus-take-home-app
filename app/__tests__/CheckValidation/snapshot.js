import { render } from "@testing-library/react";
import CheckValidation from "../../components/CheckValidation/CheckValidation";

it("renders checkvalidation unchanged", () => {
  const { container } = render(<CheckValidation />);
  expect(container).toMatchSnapshot();
});
