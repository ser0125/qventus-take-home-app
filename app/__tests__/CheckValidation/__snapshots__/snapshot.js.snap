// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checkvalidation unchanged 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <div
      class="w-10 h-10 rounded-full mx-2.5 my-1 flex justify-center bg-red-700"
    >
      <img
        alt="x_mark"
        class="w-5"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
    <span />
  </div>
</div>
`;
